### DISCUSSION TITLE
Developing some Cross-Platform OpenGL Anti-aliasing Examples

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#610] [Ideas]
Whilst building my fltk-app, I've come across the rather troublesome issue of anti-aliasing an OpenGL window.  It seems like AA can be tricky to implement in a cross-platform manner, as it is often handled by a window context, if I understand correctly. I [asked about this on the fltk discussion board, ](https://groups.google.com/g/fltkgeneral/c/mRoyUoC2-9g) and Ian recommended using GLEW to access  higher order OpenGL functions, which I assume includes options for AA.

Is there any way to use GLEW with fltk-rs, or would we need some kind of rust alternative?

Any thoughts on how to get AA working for some of the examples in the [demos ](https://github.com/MoAlyousef/fltk-rs-demos) repo?

**Comment by MoAlyousef** [comment]
I'm not an OpenGL expert, so I'm not sure if this addresses the issue.
The bindings use GLAD (via the gl_loader crate) to load opengl function pointers while using GlutWindow.
You can add the multisample mode to the GlutWindow, then call the loaded function gl_enable(GL_MULTISAMPLE).
```rust
    let app = app::App::default();
    let mut win = window::GlutWindow::new(100, 100, 500, 400, "");
    win.set_mode(Mode::Opengl3 | Mode::MultiSample);
    win.end();
    win.show();
    win.make_current();
	
    gl::load_with(|s| win.get_proc_address(s));
    unsafe { gl::Enable(gl::MULTISAMPLE); }
```

**Reply by wyhinton** [reply]
I see! Alright this helps me understand much more about how gl works in FLTK-RS. Now I feel I know where to start looking to explore options for sampling/smooth: [https://docs.rs/gl/0.14.0/gl/Enable/index.html?search=SAMPLE](https://docs.rs/gl/0.14.0/gl/Enable/index.html?search=SAMPLE)
Hope to add some examples to my demos repo soon!
